### ISSUE TITLE
Mapped tuple types won't work if they are not extracted to an type alias

### ISSUE THREAD

**Issue by devanshj** [microsoft/TypeScript#43374]
# Bug Report

### 🔎 Search Terms

Mapped tuple types

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDaBvKA3AhgGwK4QC4oAiTYqAXwBoo0s9CSAjcigXQCgPRIoAhAIwJayAJJQAlgDsoAawggA9gDNYbIgAUATorAAeGElFsaxevmIA+SlAD0tqABNFEAM5Qpi4FADuirbLc4NAAypgAthAwrjBMiugQfLjA0TDYEpiuAOqYwHoAKtaIaEaSMvJKqvnqUNq6BUYmJOYQVjb2vv6yrkG8fABMwmGRqXEJSSkx6Zk5eTCWXNgQ3kwCRILCSKTEpixsdg64UhAAHpAAxsAQjlAQWjpaHEsr-euDiFtku8ScvdAAEkJEBtTlcpI53EhpMo7lAABo0AB0yMwUhASDY+wA-PCoERjgktABuA5OFzuTzePwBP5Qf7vfiDUEQcGQ6GwhFQZGI1HozFQHFwvEeCCEkkdandRbLKAACzWdKBJDIpKOpwuVxudweTxlstedIZ2y4PGgdX0+RoAGkilArbcTmCIXIFCooPkBe6kFb9vjRXcgA)

### 💻 Code

```ts
type A = [{ value: "a" }, { value: "b" }]

type B1 = { [I in keyof A]: Prop<A[I], "value"> } // does not work
type SameAsAboveButAsAliasWat<T> = { [I in keyof T]: Prop<T[I], "value"> } // works
type B2 = SameAsAboveButAsAliasWat<A>

let b1: B1 = ["a", "b"] // unexpected error
let b2: B2 = ["a", "b"]

type H1 = B1 extends [infer X, ...any[]] ? X : never; // does not work
type H2 = B2 extends [infer X, ...any[]] ? X : never; // works

let h1: H1 = "a" // unexpected error
let h2: H2 = "a"

type Prop<T, K> = K extends keyof T ? T[K] : never
```

### 🙁 Actual behavior

Mapped tuple types behave as expected only when when extracted to a type alias.

### 🙂 Expected behavior

Mapped tuple types should behave as expected regardless of where they are extracted to a type alias or not. The above code should have no errors.

**Comment by RyanCavanaugh** [comment]
Mapped types over generic type parameters are different from mapped types over concrete types. See https://stackoverflow.com/questions/59790508/what-does-homomorphic-mapped-type-mean

**Comment by devanshj** [comment]
Ah my bad. Thanks for the link to the answer!
